byteparse/shared/src/test/scala/byteparse/ByteTests.scala

```diff
@@ -116,6 +116,27 @@ object ByteTests extends TestSuite {
 
         val Parsed.Success(Some(Array(2)), 4) = captureOpt.parse(strToBytes("01 01 01 02"))
       }
+      'unapply {
+        val capture1 = P(BS(1).rep.! ~ BS(2) ~ End)
+
+        val capture1(Array(1, 1, 1)) = strToBytes("01 01 01 02")
```

Comment:
Why do you use `Array` factory instead of `BS` or `ByteSeq` function? They are simple aliases of `Array[Byte]`, but their purpose is actually to replace explicit `Array` construction and improve readability (not only in `Parser` describing :) )